src/Nethermind/Nethermind.Core/ExecutionRequest/ExecutionRequestExtensions.cs

```diff
@@ -38,25 +38,17 @@ public static int GetRequestsByteSize(this IEnumerable<ExecutionRequest> request
     public static Hash256 CalculateHashFromFlatEncodedRequests(byte[][]? flatEncodedRequests)
     {
         // make sure that length is exactly 3
-        if (flatEncodedRequests is null || flatEncodedRequests.Length != RequestPartsCount)
+        if (flatEncodedRequests is null || flatEncodedRequests.Length > MaxRequestsCount)
         {
-            throw new ArgumentException("Flat encoded requests must be an array of 3 elements");
+            throw new ArgumentException("Flat encoded requests must be an array of 3 or less elements");
         }
 
         using SHA256 sha256 = SHA256.Create();
-        byte[] concatenatedHashes = new byte[Hash256.Size * RequestPartsCount];
-        int currentPosition = 0;
-        byte type = 0;
-        // Allocate the buffer once outside the loop
-        Span<byte> requestBuffer = stackalloc byte[Math.Max(Math.Max(flatEncodedRequests[0].Length, flatEncodedRequests[1].Length), flatEncodedRequests[2].Length) + 1];
-        // Compute sha256 for each request and concatenate them
+        using ArrayPoolList<byte> concatenatedHashes = new(Hash256.Size * MaxRequestsCount);
         foreach (byte[] requests in flatEncodedRequests)
         {
-            requestBuffer[0] = type;
-            requests.CopyTo(requestBuffer.Slice(1, requests.Length));
-            sha256.ComputeHash(requestBuffer.Slice(0, requests.Length + 1).ToArray()).CopyTo(concatenatedHashes.AsSpan(currentPosition, Hash256.Size));
-            currentPosition += Hash256.Size;
-            type++;
+            if (requests.Length <= 1) continue;
+            concatenatedHashes.AddRange(sha256.ComputeHash(requests));
```

Comment:
Why `AddRange` instead of `Add`?

Reply:
`sha256.ComputeHash(requests)` returns an array of byte